webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-student-button.js

```diff
@@ -84,6 +98,33 @@ class SakaiRubricStudentButton extends RubricsElement {
     })
     .catch (error => console.error(error));
   }
+
+  setupHidden() {
+
+    if (this.dontCheckAssociation) {
+      this.hidden = true;
+    } else {
+      this.apiGetAssociation()
+        .then(association => {
+          this.hidden = association.parameters.hideStudentPreview;
+        })
+        .catch(error => console.error(error));
+    }
+  }
+
+  apiGetAssociation() {
```

Comment:
You don't need to add this here. Use the rubricsApiMixin. Take a look at sakai-rubric-grading.js for an example.

Reply:
I tried this from the beginning, actually. The reason I added it here manually was that I couldn't get any access to rubricsApiMixin...sakai-rubric-student-button already extends RubricsElement/rubrics-element.js, and it wouldn't compile if I tried to make it extend both. If there's another way I can get access to the mixin methods, I'd love to know it!

Reply:
That's odd. You should just be able to wrap the mixin around RubricElement and it work. I'll give it a try.

Reply:
I just got it based on your last reply! I had been making a mistake with the syntax.
Here's what I just realized based on what you said:
_class SakaiRubricStudentButton extends rubricsApiMixin(RubricsElement)_
Here's what I had been trying earlier:
_class SakaiRubricStudentButton extends RubricsElement, rubricsApiMixin(RubricsElement)_